crates/pathfinder/src/storage/state.rs

```diff
@@ -1183,6 +1184,22 @@ impl StarknetStateUpdatesTable {
 
         Ok(Some(state_update))
     }
+
+    /// Deletes all rows from __head down-to reorg_tail__
+    /// i.e. it deletes all rows where `block number >= reorg_tail`.
+    pub fn reorg(tx: &Transaction<'_>, reorg_tail: StarknetBlockNumber) -> anyhow::Result<()> {
```

Comment:
The two line documentation note could be made single line but why is this separate from starknet_blocks, when is it meant to be reorgged, if not when starknet_blocks is reorgged?

Is deletion through rowid is used because sqlite doesn't support joins on deletes?

Also wondering why not just cascade the delete with the foreign key, or is there an sqlite issue?

Reply:
Raah, cascade was already discussed in the other resolved comment but it seems outdated, I think block_hash is now the primary key.

Reply:
Yeah it is now primary key invalidating some of the earlier discussion, I was also suprised to find it from the commit with the tool.

Reply:
This `reorg()` thing is now gone, and the DB uses `ON DELETE CASCADE`.